File: src/main/java/ex_pure/CountSourceLines.java
package ex_pure;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class CountSourceLines {

    public static long countSourceLines(Path sourceFile) throws IOException {
        long counter = 0;
        List<String> allLines = Files.readAllLines(sourceFile);
        for (int i = 0; i < allLines.size(); i++) {
            String line = allLines.get(i);
            if (!line.trim().isEmpty()) {
                counter++;
            }
        }
        return counter;
    }

}

File: src/main/ts/ex_pure/CountSourceLines.js
const fs = require('fs');

export function countSourceLines(file) {
    var data = fs.readFileSync(file, 'utf8');
    var lines = data.split("\n");
    var counter = 0;
    for (var i = 0; i < lines.length; i++) {
         var line = lines.at(i);
         if (line.trim() !== "") {
              counter++;
         }
    }
    return counter;
}